linode/manager
# refactor: [M3-6281] - SRC > Features > Search CSS Migration

## Description 📝
Migrated Features > Search to MUI v5

## Major Changes 🔄
1. Remove `React.FC` or `React.FunctionComponent` and rename interface `Props` to `{ComponentName}Props`. Interface can be applied to `props` parameter. IE: (props: MyProps)
2. Removed `CombinedProps` if it's directly referencing `Props` with nothing additional
3. Removed `makeStyles` for new `styled API`. We typically use `Styled{ComponentName}` as our prefix
4. `export` component inline, and remove `export default {ComponentName}` at bottom of file.
5. Eliminated unnecessary higher order component
6. Updated test from enzyme to react testing library

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-08-10 at 5 11 27 PM](https://github.com/linode/manager/assets/139489745/70cdb66e-60e7-4c1c-924a-052cbe62dccb) | ![Screenshot 2023-08-10 at 5 11 52 PM](https://github.com/linode/manager/assets/139489745/f0bf2b48-7ce9-4148-9f3a-a428064c3717) |
| ![Screenshot 2023-08-10 at 5 10 49 PM](https://github.com/linode/manager/assets/139489745/ad66ebae-c0f0-4fe3-ad5e-06e8028b472e) | ![Screenshot 2023-08-10 at 5 10 26 PM](https://github.com/linode/manager/assets/139489745/060e82cd-92b2-41b2-92e0-7477f9837cba) |

## How to test 🧪
1. Navigate to http://localhost:3000/search
2. Ensure the 'You searched for ...' icon is the same
3. Ensure the search bar is the same
4. Type something into the search bar. Ex. "label:"
7. Ensure the search result page looks the same
